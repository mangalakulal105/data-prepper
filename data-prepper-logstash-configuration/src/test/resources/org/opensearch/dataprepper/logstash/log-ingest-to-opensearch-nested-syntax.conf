input {
    http {
        port => 3000
    }
}
filter {
    grok {
        match => ["[data][log1]", "%{COMBINEDAPACHELOG}"]
        match => ["[data][log1]", "%{NUMBER} %{GREEDYDATA:[nested][field][data1]}"]
        match => {"[data][log2]" => "%{COMBINEDAPACHELOG}"}
        match => {"[data][log2]" => "%{NUMBER:num:int} %{GREEDYDATA:[nested][field][data2]}"}
        overwrite => ["[nested][field][data1]", "[nested][field][data2]"]
        break_on_match => false
    }
    date {
        match => ["[data][timestamp]", "yyyy-MM-dd"]
        target => "[data][@timestamp]"
    }
    kv {
        source => "[data][message]"
        target => "[data][test]"
    }
    mutate {
        add_field => { "[messages][message1]" => 3 "[messages][message2]" => 4.2 "[messages][message3]" => "test3" "[messages][message4]" => true }
        rename => { "[messages][messagea]" => "[newMessage]" "[messages][messageb]" => "[newMessage2]" }
        copy => { "[messages][messagec]" => "[messaged]" }
        remove_field => ["[messages][test]", "[test2]"]
        remove_field => ["[messages][test3]", "[test4]"]
        uppercase => ["[message][uppercaseField]"]
    }
}
output {
    elasticsearch {
        hosts => ["https://localhost:19000"]
        user => myuser
        password => mypassword
        index => "simple-pipeline"
    }
}