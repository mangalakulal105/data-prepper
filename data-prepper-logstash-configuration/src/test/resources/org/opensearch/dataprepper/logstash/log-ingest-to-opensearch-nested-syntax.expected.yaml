logstash-converted-pipeline:
  source:
    http:
      max_connection_count: 500
      request_timeout: 10000
      port: 3000
  processor:
    - grok:
        break_on_match: false
        match:
          /data/log1:
            - "%{COMBINEDAPACHELOG}"
            - "%{NUMBER} %{GREEDYDATA:/nested/field/data1}"
          /data/log2:
            - "%{COMBINEDAPACHELOG}"
            - "%{NUMBER:num:int} %{GREEDYDATA:/nested/field/data2}"
        keys_to_overwrite:
          - "/nested/field/data1"
          - "/nested/field/data2"
    - date:
        from_time_received: false
        destination_timezone: "UTC"
        destination: "/data/@timestamp"
        match:
          - key: "/data/timestamp"
            patterns:
              - "yyyy-MM-dd"
    - key_value:
        source: "/data/message"
        destination: "/data/test"
    - add_entries:
        entries:
          - key: "/messages/message1"
            value: 3
          - key: "/messages/message2"
            value: 4.2
          - key: "/messages/message3"
            value: "test3"
          - key: "/messages/message4"
            value: true
    - rename_keys:
        entries:
          - from_key: "/messages/messagea"
            to_key: "/newMessage"
          - from_key: "/messages/messageb"
            to_key: "/newMessage2"
    - copy_values:
        entries:
          - from_key: "/messages/messagec"
            to_key: "/messaged"
    - delete_entries:
        with_keys:
          - "/messages/test"
          - "/test2"
          - "/messages/test3"
          - "/test4"
    - uppercase_string:
        with_keys:
          - "/message/uppercaseField"
  sink:
    - opensearch:
        hosts:
          - "https://localhost:19000"
        username: "myuser"
        password: "mypassword"
        index: "simple-pipeline"
